### PR TITLE
Add Kruskal's minimum spanning tree algorithm implementation.

### PR DESCRIPTION
**Pull request by izanbf1803**
Tested, working correctly.

### CODE REVIEW

File: cache/lru_cache.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
+class LRUCacheNode:
+  def __init__(self, k, v):
+      self.key = k
+      self.value = v
+      self.prev = None
+      self.next = None
+
```
Comment (ankit167) [inline cache/lru_cache.py]: 2 blank spaces between classes

File: cache/lru_cache.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+class LRUCache(object):
+    def __init__(self, capacity):
+        self.capacity = capacity
+        self.head = LRUCacheNode(None, 'head')
+        self.tail = LRUCacheNode(None, 'tail')
+        self.head.next = self.tail # tail being most recent
```
Comment (ankit167) [inline cache/lru_cache.py]: Atleast two spaces before inline comment. Line 13,14

File: cache/lru_cache.py [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+        self.tail.prev = self.head # head being oldest
+        self.data = {}
+
+    def deleteNode(self,node):
```
Comment (ankit167) [inline cache/lru_cache.py]: Add space after ','. Line 17, 24, 41, 50

Comment (ankit167) [inline cache/lru_cache.py]: Snake case recommended for function/class name

File: cache/lru_cache.py [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+        self.insertNew(newNode)
+
+    def insertNew(self,newNode):
+        # insert new node into last position    
+        last = self.tail.prev
+        last.next = newNode
+        self.tail.prev = newNode
+        newNode.next = self.tail
+        newNode.prev = last
```
Comment (ankit167) [inline cache/lru_cache.py]: Please add sample test case inside main()

File: cache/lru_cache.py [hunk 5]
```diff
@@ -0,0 +1,56 @@
+class LRUCacheNode:
```
Comment (ankit167) [inline cache/lru_cache.py]: Please add brief overview of the LRU implementation. It will be easier for folks to understand the code.

**Comment by izanbf1803** [comment]
I deleted my implementation of the LRU Cache because i don't have experience with that DS.
I also implemented the Kruskal algorithm for the minimum spanning tree, and i have explained my implementation (common implementation).

**Comment by keon** [comment]
@izanbf1803 Thanks!
We might have to consolidate the comment style but that could go onto other PR.  :)

**Pull request merged**